File: lab1/src/main/kotlin/com/itmo/blss/model/jaas/RolePrincipal.java
package com.itmo.blss.model.jaas;

import java.security.Principal;

import javax.security.auth.Subject;

public record RolePrincipal(
        String name
) implements Principal {

    @Override
    public String getName() {
        return name;
    }

    @Override
    public boolean implies(Subject subject) {
        return Principal.super.implies(subject);
    }
}

File: lab1/src/main/kotlin/com/itmo/blss/model/jaas/RolePrincipal.kt
package com.itmo.blss.model.jaas

import java.security.Principal
import javax.security.auth.Subject

@JvmRecord
data class RolePrincipal(val name: String) : Principal {
    override fun getName(): String {
        return name
    }

    override fun implies(subject: Subject): Boolean {
        return super<Principal>.implies(subject)
    }
}